hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/model/EventTimeFlinkRecordMerger.java

```diff
@@ -50,19 +65,24 @@ public Option<Pair<HoodieRecord, Schema>> merge(
     ValidationUtils.checkArgument(newer.getRecordType() == HoodieRecord.HoodieRecordType.FLINK);
 
     if (older.getOrderingValue(oldSchema, props).compareTo(newer.getOrderingValue(newSchema, props)) > 0) {
-      if (older.isDelete(oldSchema, props)) {
+      if (older.isDelete(oldSchema, props) && !isPreCombineMode) {
```

Comment:
There is no need to distinguish the `isPreCombineMode`, just returns the record instead of `Option.empty`.